File: ssr.js
import fs from 'fs/promises'; // Use fs with promises
import { JSDOM } from 'jsdom';

const wrapperPath = 'wrapper.html'; // Path to wrapper template
const blogPostPath = 'renderedPosts/ssrTesting.html'; // Path to blog post content
const outputPath = 'index.html'; // Output file path

const renderHTML = async () => {
  try {
    // Step 1: Read the wrapper template
    console.log('wrapperHtml')
    const wrapperHtml = await fs.readFile(wrapperPath, 'utf8');

    // Step 2: Parse the wrapper into a DOM object
    const wrapperDom = new JSDOM(wrapperHtml);
    const wrapperDocument = wrapperDom.window.document;

    // Step 3: Read the blog post content
    const blogHtml = await fs.readFile(blogPostPath, 'utf8');

    // Step 4: Parse the blog post into a DOM object
    const blogDom = new JSDOM(blogHtml);
    const blogBodyContent = blogDom.window.document.body.innerHTML; // Extract blog content

    // Step 5: Insert blog post content into the wrapper
    const contentDiv = wrapperDocument.querySelector('.content'); // Locate the placeholder div
    if (contentDiv) {
      contentDiv.innerHTML = blogBodyContent; // Replace content
    } else {
      console.error('Error: Could not find .main div in the wrapper.');
      return;
    }

    // Step 6: Serialize and write the final HTML to output
    const finalHtml = wrapperDom.serialize();
    await fs.writeFile(outputPath, finalHtml, 'utf8');
    console.log(`SSR complete. Rendered HTML saved to ${outputPath}.`);
  } catch (err) {
    console.error(`Error: ${err.message}`);
  }
};

renderHTML();

File: ssr.py
from bs4 import BeautifulSoup

"""
This is currently broken.

bs4 is annoying because it's not clear what all of the types are:
I would rather strictly use css selectors and deal with strings and lists
"""

# File paths
wrapper_path = "wrapper.html"  # Path to wrapper template
blog_post_path = "renderedPosts/ssrTesting.html"  # Path to blog post content
output_path = "index.html"  # Output file

# Step 1: Read the wrapper template
with open(wrapper_path, "r") as wrapper_file:
  wrapper_html = wrapper_file.read()
wrapper_soup = BeautifulSoup(wrapper_html, "html.parser")  # Parse into DOM object

# Step 2: Read the blog post content
with open(blog_post_path, "r") as blog_file:
  blog_html = blog_file.read()
blog_post_soup = BeautifulSoup(blog_html, "html.parser")  # Parse into DOM object

# Step 3: Replace the inner content of the content div with the blog post
main_div = wrapper_soup.select_one('.main')  # Locate placeholder div
if main_div:
  main_div.clear()  # Clear existing content
  children = blog_post_soup.select_one('body > *')
  for child in children:
    breakpoint()
    main_div.append(child)
  #main_div.append(blog_post_soup.select_one('body'))

# Step 4: Write the final rendered HTML to a file
with open(output_path, "w") as output_file:
  output_file.write(str(wrapper_soup))  # Serialize and save the DOM as HTML

# Debug message
print(f"SSR complete. Rendered HTML saved to {output_path}.")
